packages/m2m-gateway/test/api/agreements/rejectAgreement.test.ts

```diff
@@ -0,0 +1,106 @@
+import { describe, it, expect, vi } from "vitest";
+import { generateToken } from "pagopa-interop-commons-test";
+import { AuthRole, authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { agreementApi, m2mGatewayApi } from "pagopa-interop-api-clients";
+import { api, mockAgreementService } from "../../vitest.api.setup.js";
+import { appBasePath } from "../../../src/config/appBasePath.js";
+import {
+  missingMetadata,
+  resourcePollingTimeout,
+} from "../../../src/model/errors.js";
+import { getMockedApiAgreement } from "../../mockUtils.js";
+import { toM2MGatewayApiAgreement } from "../../../src/api/agreementApiConverter.js";
+
+describe("POST /agreements/:agreementId/reject router test", () => {
+  const mockApiAgreement = getMockedApiAgreement({
+    state: agreementApi.AgreementState.Values.PENDING,
+  });
+
+  const mockRejectAgreementBody: m2mGatewayApi.AgreementRejection = {
+    reason: "This is a test reason for rejection",
+  };
+
+  const mockM2MAgreementResponse: m2mGatewayApi.Agreement =
+    toM2MGatewayApiAgreement(mockApiAgreement.data);
+
+  const makeRequest = async (
+    token: string,
+    body: m2mGatewayApi.AgreementRejection,
+    agreementId: string = mockApiAgreement.data.id
+  ) =>
+    request(api)
+      .post(`${appBasePath}/agreements/${agreementId}/reject`)
+      .set("Authorization", `Bearer ${token}`)
+      .send(body);
+
+  const authorizedRoles: AuthRole[] = [authRole.M2M_ADMIN_ROLE];
+  it.each(authorizedRoles)(
+    "Should return 200 and perform service calls for user with role %s",
+    async (role) => {
+      mockAgreementService.rejectAgreement = vi
+        .fn()
+        .mockResolvedValue(mockM2MAgreementResponse);
+
+      const token = generateToken(role);
+      const res = await makeRequest(token, mockRejectAgreementBody);
+
+      expect(res.status).toBe(200);
+      expect(res.body).toEqual(mockM2MAgreementResponse);
+    }
+  );
+
+  it.each([missingMetadata(), resourcePollingTimeout(3)])(
+    "Should return 500 in case of $code error",
+    async (error) => {
+      mockAgreementService.rejectAgreement = vi.fn().mockRejectedValue(error);
+      const token = generateToken(authRole.M2M_ADMIN_ROLE);
+      const res = await makeRequest(token, mockRejectAgreementBody);
+
+      expect(res.status).toBe(500);
+    }
+  );
+
+  it("Should return 400 for incorrect value for purpose id", async () => {
+    mockAgreementService.rejectAgreement = vi
+      .fn()
+      .mockResolvedValue(mockM2MAgreementResponse);
+
+    const token = generateToken(authRole.M2M_ROLE);
+    const res = await makeRequest(token, mockRejectAgreementBody, "INVALID ID");
+    expect(res.status).toBe(400);
+  });
+
+  it.each([
+    { ...mockRejectAgreementBody, invalidParam: "invalidValue" },
+    { ...mockRejectAgreementBody, reason: undefined },
```

Comment:
```suggestion
    { reason: undefined },
```

No need to spread siince the body has just the reason property